sql/schema_changer.go

```diff
@@ -54,3 +59,77 @@ func (p *planner) applyMutations(tableDesc *TableDescriptor) error {
 	p.notifyCompletedSchemaChange(newTableDesc.ID)
 	return nil
 }
+
+// TableLeaderLeaser is used to acquire a table leader lease on a table.
+type TableLeaderLeaser struct {
+	p *planner
+}
+
+// NewTableLeaderLeaserForTesting for tests.
+func NewTableLeaderLeaserForTesting(txn *client.Txn) TableLeaderLeaser {
+	return TableLeaderLeaser{p: &planner{txn: txn}}
+}
+
+var errExistingTableLeader = errors.New("an outstanding lease exists")
+
+// Acquire acquires a leader lease on the table if
+// an unexpired lease doesn't exist. It returns the lease.
+func (t TableLeaderLeaser) Acquire(tableID ID) (TableDescriptor_LeaderLease, error) {
+	t.p.txn.SetSystemDBTrigger()
+	tableDesc, err := t.p.getTableDescFromID(tableID)
+	if err != nil {
+		return TableDescriptor_LeaderLease{}, err
+	}
+
+	now := time.Now()
+	// Add 5 seconds to the lease expiration to deal with time uncertainty across nodes.
+	if tableDesc.Lease != nil && time.Unix(tableDesc.Lease.ExpirationTime+5, 0).After(now) {
+		// Return the existing lease with an error.
+		return *tableDesc.Lease, errExistingTableLeader
+	}
+
+	leader := parser.GenerateUniqueBytes(t.p.evalCtx.NodeID)
+	lease := TableDescriptor_LeaderLease{Leader: string(leader), ExpirationTime: now.Add(jitteredLeaseDuration()).Unix()}
+	tableDesc.Lease = &lease
+	err = t.p.txn.Put(MakeDescMetadataKey(tableDesc.ID), wrapDescriptor(tableDesc))
+	return lease, err
+}
+
+func (p *planner) findTableWithLease(tableID ID, lease TableDescriptor_LeaderLease) (*TableDescriptor, error) {
+	tableDesc, err := p.getTableDescFromID(tableID)
+	if err != nil {
+		return nil, err
+	}
+	if tableDesc.Lease == nil {
+		return nil, util.Errorf("no lease present for tableID: %d", tableID)
+	}
+	if tableDesc.Lease.Leader != lease.Leader {
+		return nil, util.Errorf("lease for table ID: %d, owned by different leader: %q, expected: %q", tableID, tableDesc.Lease.Leader, lease.Leader)
```

Comment:
The strings in `Lease.Leader` are opaque, so it doesn't make much sense to print them. I think it would be good for debugability to store the leader's node ID explicitly in the lease (and then a lease instance should be uniquely identifiable by the node ID and expiration time)